File: Code/Components/DanpheEMR.ServerModel/AccountingModels/VoucherLedgerGroupMapModel.cs

using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace DanpheEMR.ServerModel
{
    public class VoucherLedgerGroupMapModel
    {      
        [Key]
        public int VoucherLedgerGroupMapId { get; set; }
        public int VoucherId { get; set; }
        public int LedgerGroupId { get; set; }
        public bool IsDebit { get; set; }
        public DateTime? CreatedOn { get; set; }
        public int CreatedBy { get; set; }
        public bool IsActive { get; set; }

        [NotMapped]
        public string actionName { get; set; }
    }
}

File: Code/Websites/DanpheEMR/wwwroot/DanpheApp/src/app/accounting/settings/shared/voucher-ledger-group-map.model.ts
﻿import {
    FormBuilder,
    FormGroup
} from '@angular/forms';
import * as moment from 'moment';
import { ENUM_DateTimeFormat } from '../../../shared/shared-enums';

export class VoucherLedgerGroupMapModel {
    public VoucherLedgerGroupMapId: number = 0;
    public VoucherId: number = 0;
    public LedgerGroupId: number = 0;
    public IsDebit: boolean = false;
    public CreatedBy: number = 0;
    public CreatedOn: string = null;
    public IsActive: boolean = true;
    public flagCredit: boolean = true;
    public flagDebit: boolean = true;
    public voucherName: string = null;
    public isMapped: boolean = false;
    public actionName: string = null;
    public VoucherCode: string = null;
    public VoucherLedgerGroupMapValidator: FormGroup = null;
    constructor() {
        this.CreatedOn = moment().format(ENUM_DateTimeFormat.Year_Month_Day_Hour_Minute);
        var _formBuilder = new FormBuilder();
        this.VoucherLedgerGroupMapValidator = _formBuilder.group({
            ///// 'VoucherName': ['', Validators.compose([Validators.required])],
        });
    }

    public IsDirty(fieldName): boolean {
        if (fieldName == undefined)
            return this.VoucherLedgerGroupMapValidator.dirty;
        else
            return this.VoucherLedgerGroupMapValidator.controls[fieldName].dirty;
    }

    public IsValid(): boolean { if (this.VoucherLedgerGroupMapValidator.valid) { return true; } else { return false; } } public IsValidCheck(fieldName, validator): boolean {
        if (fieldName == undefined) {
            return this.VoucherLedgerGroupMapValidator.valid;

        }

        else
            return !(this.VoucherLedgerGroupMapValidator.hasError(validator, fieldName));
    }
}